apps/web-staking/src/app/hooks/useGetRedemptions.ts

```diff
@@ -4,61 +4,79 @@ import { useAccount } from 'wagmi';
 import { getAllRedemptions, OrderedRedemptions, refreshRedemptions } from '@/services/redemptions.service';
 import useSessionStorage from './useSessionStorage';
 
+const INITIAL_REDEMPTIONS: OrderedRedemptions = {
+    claimable: [],
+    open: [],
+    closed: [],
+};
+
 const useGetRedemptions = () => {
-    const { address, chainId } = useAccount();
+    const { address, chainId, isConnected } = useAccount();
+    const [loadedAddress, setLoadedAddress] = useState<`0x${string}` | undefined>(undefined);
```

Comment:
If we would save the redemptions to a wallet specific sessions storage like session.get(`userRedemptions-${address}`), could we not get rid of the `loadedAddress` logic here ?
This would also make sure we never overwrite or merge sessions of different connected wallets without having to actually implement anything.

Reply:
I have updated this. I had to also update useSessionSessionStorage to trigger a re-render when the storedValue/key changes.